libntfs-3g/mft.c

```diff
@@ -1936,13 +1935,6 @@ ntfs_inode *ntfs_mft_record_alloc(ntfs_volume *vol, ntfs_inode *base_ni)
 	if (oldwarn)
 		NVolClearNoFixupWarn(vol);
 
-	/* Sanity check that the mft record is really not in use. */
-	if (ntfs_is_file_record(m->magic) && (m->flags & MFT_RECORD_IN_USE)) {
-		ntfs_log_error("Inode %lld is used but it wasn't marked in "
-			       "$MFT bitmap. Fixed.\n", (long long)bit);
-		free(m);
-		goto retry;
-	}
```

Comment:
I think that this code is cause other issues. I think having this code is rather problematic. Because this code sets the bits while finding unused record. This can create an orphan mft entry. Currently ntfsck does not check the record unless the mft bitmap bit is set. This code will cause a difference in behavior. I think that it is correct to handle records for which the mft bitmap bit is not set as deleted files.